include/swift/Driver/ExperimentalDependencyDriverGraph.h

```diff
@@ -0,0 +1,338 @@
+//===--- ExperimentalDependencyDriverGraph.h --------------------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef ExperimentalDependencyGraph_h
+#define ExperimentalDependencyGraph_h
+
+#include "swift/Basic/ExperimentalDependencies.h"
+#include "swift/Basic/LLVM.h"
+#include "swift/Basic/OptionSet.h"
+#include "swift/Driver/DependencyGraph.h"
+#include "swift/Driver/Job.h"
+#include "llvm/ADT/ArrayRef.h"
+#include "llvm/ADT/DenseMap.h"
+#include "llvm/ADT/STLExtras.h"
+#include "llvm/ADT/SmallPtrSet.h"
+#include "llvm/ADT/StringMap.h"
+#include "llvm/ADT/StringRef.h"
+#include "llvm/ADT/StringSet.h"
+#include "llvm/ADT/iterator_range.h"
+#include "llvm/Support/PointerLikeTypeTraits.h"
+#include <string>
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+
+// Declarations for the portion experimental dependency system used by the
+// driver.
+
+namespace swift {
+namespace driver {
+namespace experimental_dependencies {
+
+//==============================================================================
+// MARK: Cross-references to names not in driver namespace
+//==============================================================================
+
+// Awkward... needs to be in driver and experimental_dependencies, but the
```

Comment:
I think it'd be okay to just use swift::experimental_dependencies if you think it makes things simpler. We don't use namespaces for every subsystem in the compiler.

Reply:
Thanks. Done.